core/lib/libfdt/fdt_rw.c

```diff
@@ -285,7 +285,8 @@ int fdt_setprop(void *fdt, int nodeoffset, const char *name,
 	if (err)
 		return err;
 
-	memcpy(prop->data, val, len);
+	if (len)
+		memcpy(prop->data, val, len);
```

Comment:
OK so the problem is with `val == NULL`, not with `len == 0` as I thought. Indeed the pointers supplied to `memcpy()` have to be valid even when `len == 0`.
Should it be: `if (len) { assert(val); mempcy(...); }` instead?
Also, I suggest you update the commit comment with what you wrote above.

In any case: `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`.